hibike/devices/BatteryBuzzer/battery_buzzer.cpp

```diff
@@ -1,323 +1,134 @@
-  TIFR1 = (1<<OCF1A);  // Clear previous interrupt flag
-  TIMSK1 = (1<<OCIE1A);  // Enable interrupt on compare A
-  sei();
+  return true; // device is calibrated (no default values)
 }
 
-// Run at 3125 Hz (every 320 us)
-ISR(TIMER1_COMPA_vect) {
-  uint16_t nextVal = ADC;  // Assume that the previous conversion finished
-  // Start next analog read
-  ADMUX = (ADMUX & 0xF8) | (channelNums[inputIndex] & 0x07);  // Select ADC channel
-  ADCSRA |= (1<<ADSC);  // Start conversion
-  // Accumulate inputs so an average analog input can be generated
-  if(lastInputIndex == inputIndex) {
-    // This is the first time
-    // there is no data yet
-    inputIndex++;
-    // Zero the new accumulation array
-    memset(analogAccumArrs[!latestAnalogAccum], 0, NUM_CELLS*sizeof(uint16_t));
-    disconnectedArr[!latestAnalogAccum] = 0;
-  } else {
-    analogAccumArrs[!latestAnalogAccum][lastInputIndex] += nextVal;  // Write data to the other output array
-    disconnectedArr[!latestAnalogAccum] += nextVal < DISCONNECTED_THRESHOLD;
-    if (disconnectedArr[!latestAnalogAccum] > DISCONNECTED_COUNT) {
-      disconnectedArr[!latestAnalogAccum] = DISCONNECTED_COUNT;
-    }
-    if (inputIndex == 0 && inputSample == 0) {
-      uint8_t arrIndex = !latestAnalogAccum;
-      cellVoltageArrs[arrIndex][0] = analogAccumArrs[arrIndex][0]*cellScale[0];
-      cellVoltageArrs[arrIndex][1] = analogAccumArrs[arrIndex][1]*cellScale[1];
-      cellTotalArr[arrIndex] = analogAccumArrs[arrIndex][2]*cellScale[2];
-      cellVoltageArrs[arrIndex][2] = cellTotalArr[arrIndex] - cellVoltageArrs[arrIndex][1];
-      cellVoltageArrs[arrIndex][1] -= cellVoltageArrs[arrIndex][0];
-      
-      // Just finished taking samples, output the total now
-      latestAnalogAccum = !latestAnalogAccum;
-      // Zero the new accumulation array
-      memset(analogAccumArrs[!latestAnalogAccum], 0, NUM_CELLS*sizeof(uint16_t));
-      disconnectedArr[!latestAnalogAccum] = 0;
-      measureCounter++;
-      if (measureCounter >= BLINK_AFTER_MEASURE) measureCounter = 0;
-      if (measureCounter == 0) {
-        digitalWrite(BUZZER_LED_PIN, HIGH);
-      }
-      if (measureCounter == 1) {
-        digitalWrite(BUZZER_LED_PIN, LOW);
-      }
+// You must implement this function.
+// It is called when the device receives a Device Data Update packet.
+// Modifies data to contain the parameter value.
+//    param           -   Parameter index
+//    data -   buffer to return data in, little-endian
+//    len         -   Maximum length of the buffer
+//
+//    return          -   sizeof(param) on success; 0 otherwise
 
-      // Update states and turn on/off buzzer
-      updateCellStates();
-      updateSafety();
-      lastMeasureTime = millis();
-    }
-    lastInputIndex = inputIndex;
-    if (inputIndex < NUM_CELLS-1) {
-      inputIndex++;
+uint8_t device_read(uint8_t param, uint8_t* data, size_t len) {
+  if (MAX_PAYLOAD_SIZE - len > sizeof(bool)) {
+    if(param == 0){
```

Comment:
Wasn't there an enum for the param numbers at some point? I think it's good to have one.